diff --git a/org/apache/commons/fileupload/FileUploadBase.java b/org/apache/commons/fileupload/FileUploadBase.java
index 1f64db21efcfe..01c1b1b772c02 100644
--- a/org/apache/commons/fileupload/FileUploadBase.java
+++ b/org/apache/commons/fileupload/FileUploadBase.java
@@ -1122,8 +1122,14 @@ public abstract class FileUploadBase {
         }
 
         private long getContentLength(FileItemHeaders pHeaders) {
+            String header = pHeaders.getHeader(CONTENT_LENGTH);
+
+            if ((header == null) || (header.length() == 0)) {
+                return -1;
+            }
+
             try {
-                return Long.parseLong(pHeaders.getHeader(CONTENT_LENGTH));
+                return Long.parseLong(header);
             } catch (Exception e) {
                 return -1;
             }
@@ -1535,3 +1541,4 @@ public abstract class FileUploadBase {
     }
 
 }
+/* @generated */
\ No newline at end of file
